scheduler/test/cook/test/compute_cluster.clj

```diff
@@ -0,0 +1,956 @@
+             (get-db-config-ents (d/db conn)))))))
+
+(deftest test-compute-cluster->compute-cluster-config
+  (let [cluster {:name "name"
+                 :cluster-definition {:factory-fn 'cook.kubernetes.compute-cluster/factory-fn
+                                      :config {:name "name-starting"
+                                               :template "template"
+                                               :base-path "base-path"
+                                               :ca-cert "ca-cert"
+                                               :state :running
+                                               :state-locked? false}}
+                 :state-atom (atom :deleted)
+                 :state-locked?-atom (atom true)}]
+    (is (= {:base-path "base-path"
+            :ca-cert "ca-cert"
+            :name "name"
+            :state :deleted
+            :state-locked? true
+            :template "template"}
+           (compute-cluster->compute-cluster-config cluster)))))
+(def expected-in-mem-config
+  {"name" {:base-path "base-path"
+           :ca-cert "ca-cert"
+           :name "name"
+           :state :deleted
+           :state-locked? true
+           :template "template"}})
+(def sample-clusters
+  {"name" {:name "name"
+           :cluster-definition {:factory-fn 'cook.kubernetes.compute-cluster/factory-fn
+                                :config {:name "name"
+                                         :template "template"
+                                         :base-path "base-path"
+                                         :ca-cert "ca-cert"
+                                         :state :running
+                                         :state-locked? false}}
+           :state-atom (atom :deleted)
+           :state-locked?-atom (atom true)
+           :dynamic-cluster-config? true}
+   "name-no-state" {:name "name"
+                    :cluster-definition {:factory-fn 'cook.kubernetes.compute-cluster/factory-fn
+                                         :config {:name "name"
+                                                  :template "template"
+                                                  :base-path "base-path"
+                                                  :ca-cert "ca-cert"
+                                                  :state :running
+                                                  :state-locked? false}}}})
+(deftest test-in-mem-configs
+  (reset! cluster-name->compute-cluster-atom {})
+  (is (= {} (get-in-mem-configs)))
+  (reset! cluster-name->compute-cluster-atom sample-clusters)
+  (is (= expected-in-mem-config (get-in-mem-configs))))
+
+(deftest test-get-dynamic-clusters
+  (reset! cluster-name->compute-cluster-atom {})
+  (is (= {} (get-in-mem-configs)))
+  (reset! cluster-name->compute-cluster-atom sample-clusters)
+  (is (= (->> [(sample-clusters "name")] (map-from-vals #(-> % :name)))
+         (get-dynamic-clusters))))
+
+(deftest test-compute-current-configs
```

Comment:
May want to consider more realistic dictionary shapes in some of these functions. E.g., strings instead of keywords. (I'm pretty sure that that would have identified a bug i pointed out elsewhere in review.)

Reply:
I think the shapes are correct. Do you have a counter example?